src/domain/relation/establish.ts

```diff
@@ -14,5 +18,24 @@ export default async function establish(requester: Requester, followingId: strin
         throw new RelationAlreadyExists();
     }
 
-    return create(requester.id, followingId);
+    let relation, followerCount, followingCount;
+
+    try
+    {
+        relation = await create(requester.id, followingId);
+        followerCount = await increaseFollowerCount(followingId);
+        followingCount = await increaseFollowingCount(requester.id);
+
+    }
+    catch (error: unknown)
+    {
+        const relationId = relation?.id ?? '';
+        const undoRelation = relation !== undefined ? removeRelation(relationId) : Promise.resolve();
+        const undoFollowerCount = followerCount !== undefined ? decreaseFollowerCount(followingId) : Promise.resolve();
```

Comment:
The increase functions both return a void, thus the counts are always undefined. As far as I can see, the decrease functions are never called.